chef-zero.gemspec

```diff
@@ -25,6 +21,16 @@ Gem::Specification.new do |s|
   s.add_dependency "rackup", "~> 2.2", ">= 2.2.1"
   s.add_dependency "unf_ext", "~> 0.0.8"
   s.add_dependency "webrick"
+  s.add_dependency "activesupport", "~> 7.1.5"
+  # We are running into some challenging things with activesupport and fiddle.
+  # Chef-18 requires Ruby 3.1 and Chef-19 requires Ruby 3.4. They have incompatible dependencies on activesupport and fiddle
+  # Activesupport 7.1.3.2 also has a CVE in it which requires an upgrade to 7.1.5.2. Activesupport 7.1.5 requires fiddle = 1.1.0, but Chef-19 requires fiddle >= 1.1.6.
+  # Also, fiddle is a built-in gem for Ruby 3.1 but is broken out into a separate gem starting in Ruby 3.2.
+  if Gem::Version.new(RUBY_VERSION) <= Gem::Version.new("3.1.7")
+    s.add_dependency "fiddle", "= 1.1.0"
```

Comment:
you can't do an if statement in a gemspec

Reply:
More specifically, since a gemspec is evaluated at `gem build`  time, and chef-zero is included (at least by Chef) as a gem published to RubyGems, this dependency will match the setup of the machine that built the gem, not the setup of the machine _including_ the gem. (We _do_ have an if or two in gemspecs that aren't subject to this pitfall, regardless of how dubious the practice may be.) cc: @johnmccrae